libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginListener.java

```diff
@@ -55,6 +55,7 @@
     void saveCredentialsInSmartLock(@Nullable String username, @Nullable String password,
                                     @NonNull String displayName, @Nullable Uri profilePicture);
     void loggedInViaUsernamePassword(ArrayList<Integer> oldSitesIds);
+    void loggedInToSelfHostedWebsite(ArrayList<Integer> oldSitesIds, int siteId);
```

Comment:
👋 @khaykov ! Unfortunately, I'm not a big fun of adding a new callback to the `LoginListener`. Weird as it might sound, reason is that this listener pretty much describes the different login flows and their needs in term of implementation "endpoints". Besides, this listener is defined in the login library, which means that its clients will need to implement a form of this `loggedInToSelfHostedWebsite`, including the WooCommerce Android app that is a client.

Judging from the code in the PR, it seems that the functionality of this new callback is only different to `loggedInViaUsernamePassword` by the `siteId` passed, to make it easy to select the newly added site. I wonder, why not use `loggedInViaUsernamePassword` for that? At this point, since the `oldSitesIds` list is already passed to `loggedInViaUsernamePassword`, it's only a matter of diffing against the latest sites list (from the SiteStore) to get the newly added site. That was pretty much the original reason for introducing the `oldSitesIds`, to detect the new site(s) added.

What do you think about revising the implementation to do such diffing in LoginActivity and pass the detected id to WPMainActivity?

Reply:
Thanks for taking a look and providing all the context, @hypest ! That's exactly the type of response I was looking for :) Everything you mentioned makes sense now, and I changed the code to make use of `oldSitesIds` when finding an ID of newly-added self-hosted site.